### DISCUSSION TITLE
can we teach arch.exe to use windows terminal as default terminal?

### DISCUSSION THREAD

**Discussion by WingT** [yuk7/ArchWSL#178] [Q&A] [answered]
As for now, arch.exec opens an old ugly terminal window, missing font, can we use windows terminal or any other terminal instead?

**Comment by escape0707** [comment]
Of course, Windows Terminal should automatically adds this WSL option.
If not, add this to Windows Terminal's config:
```json
            {
                "guid": "{a5a97cb8-8961-5535-816d-772efe0c6a3f}",
                "hidden": false,
                "name": "Arch",
                "source": "Windows.Terminal.Wsl"
            },
```
You can also run Windows Terminal with powershell, then execute `arch`.

**Reply by escape0707** [reply]
Oops, I see you mean to change the terminal `arch.exe` want to launch for you?

**Reply by escape0707** [reply]
I think the terminal used by `arch.exe` is the same as you Win+R and run `ping -t google.com` and any other console applications. It's `conhost` on windows by default. You should search for how to set another default terminal on windows 10, for example `ConEmu`.

**Reply by escape0707** [reply]
https://github.com/microsoft/terminal/issues/492
It’s currently no possible for Windows Terminal. Try ConEmu.

**Reply by WingT** [reply]
I found an option --set-default-term which suits my needs perfectly

**Reply by escape0707** [reply]
Oh, great! I didn’t know we actually have that option.

**Comment by tonyxwz** [comment, marked as answer]
https://github.com/yuk7/wsldl#set-windows-terminal-as-default-terminal
